plugins/github-actions/src/components/Router.tsx

```diff
@@ -20,18 +20,47 @@ import { rootRouteRef, buildRouteRef } from '../plugin';
 import { WorkflowRunDetails } from './WorkflowRunDetails';
 import { WorkflowRunsTable } from './WorkflowRunsTable';
 import { GITHUB_ACTIONS_ANNOTATION } from './useProjectName';
-import { WarningPanel } from '@backstage/core';
+import { EmptyState } from '@backstage/core';
+import { Button } from '@material-ui/core';
+import PostAddIcon from '@material-ui/icons/PostAdd';
 
 export const isPluginApplicableToEntity = (entity: Entity) =>
   Boolean(entity.metadata.annotations?.[GITHUB_ACTIONS_ANNOTATION]);
 
+const ActionButton = ({ entity }: { entity: Entity }) => {
+  const [type, target] =
+    entity.metadata.annotations?.['backstage.io/managed-by-location'].split(
+      /:(.+)/,
+    ) || [];
+
+  return type === 'github' ? (
+    <Button
+      variant="contained"
+      color="primary"
+      href={target.replace('blob', 'edit')}
+      startIcon={<PostAddIcon />}
+    >
+      Add annotation
+    </Button>
+  ) : (
+    <Button
+      variant="contained"
+      color="primary"
+      href="https://backstage.io/docs/features/software-catalog/well-known-annotations"
+    >
+      Read more
+    </Button>
+  );
+};
+
 export const Router = ({ entity }: { entity: Entity }) =>
-  // TODO(shmidt-i): move warning to a separate standardized component
   !isPluginApplicableToEntity(entity) ? (
-    <WarningPanel title="GitHubActions plugin:">
-      <pre>{GITHUB_ACTIONS_ANNOTATION}</pre> annotation is missing on the
-      entity.
-    </WarningPanel>
+    <EmptyState
+      missing="field"
+      title="Missing Annotation"
+      description={`The "${GITHUB_ACTIONS_ANNOTATION}" annotation is missing on "${entity.metadata.name}". You need to add the annotation to your component if you want to enable Github Actions for it.`}
```

Comment:
@marcuseide @samiramkr since this is a very common case, and something we want to make sure is consistent between entities, I think we should have a higher level abstraction here. Something like a `<MissingAnnotationEmptyState>` where the the only input prop is the specific `annotation` that the plugin expects.

To make it super clear to the user what they should do, I think we should even render example YAML like:

![Screenshot 2020-10-01 at 10 37 27](https://user-images.githubusercontent.com/190856/94787124-25600f80-03d2-11eb-89f6-6e4769705399.png)

Reply:
A **future improvement** could be to have a button that generates a PR that adds the specific annotation.

Reply:
@stefanalund Sounds good! 👍  Do you think the current way we do it (opening the component yaml in `edit` mode if on github) is something we should keep until we can raise PRs?

Reply:
Yep! That feels like an excellent initial flow.

Reply:
This is how it looks now:
<img width="1740" alt="Screenshot 2020-10-02 at 13 48 05" src="https://user-images.githubusercontent.com/15106494/94922944-5cf6b680-04bb-11eb-8a1f-4a734dae74fc.png">

Reply:
I love it!